src/gt4py/backend/gt_backends.py

```diff
@@ -540,8 +526,20 @@ def generate(self) -> Type["StencilObject"]:
         )
 
     def generate_computation(self) -> Dict[str, Union[str, Dict]]:
-        dir_name = f"{self.builder.options.name}_src"
-        src_files = self.make_extension_sources()
+        src_files = self.builder.backend_data.get("computation_src", None)
+        if not src_files:
+            dir_name = f"{self.builder.options.name}_src"
+            src_files = self.make_extension_sources()
+        return {dir_name: {k: v for k, v in src_files.items() if not k.startswith("bindings")}}
```

Comment:
I agree. I will modify the pyext generator to separate them in it's output. The output will still be cached at the backend because the generator does everything in one pass and that shouldn't need to change. But the overall code will look more sane and be more robust.